Lib/xml/etree/ElementPath.py

```diff
@@ -65,7 +65,7 @@
     r"//?|"
     r"\.\.|"
     r"\(\)|"
-    r"[/.*:\[\]\(\)@=])|"
+    r"[/.,*:\[\]\(\)@=])|"
```

Comment:
There does not seem to be a precedence for such testing. Is there a similar test for unusually placed '@' or '=' tokens? I could implement a test such as these:
```
        self.assertRaisesRegex(KeyError, ',', e.find, ',/tag[0]')
        self.assertRaisesRegex(KeyError, ',', e.find, '.,tag[0]')
        self.assertRaisesRegex(KeyError, ',', e.find, './,tag[0]')
        self.assertRaisesRegex(KeyError, ',', e.find, '.,/tag[0]')
```
If I added these tests, wouldn't it be necessary to add similar tests for the other characters in the regex? Do I test for every character in the regex? How far do I go?

Reply:
Most of the other characters are part of the general tree structure and thus allowed in more places. All of these places are unlikely to be prepared for handling a `,`, so this is a new case. I'd specifically recommend to add tests for things like `,[0]`, `tag[,]`, `tag[(,)]` and so on.

I would generally say that covering the parser with more tests would be helpful, but you don't have to do that for this ticket. Since you're at it, though, feel free to add a couple of similar ones for `=`, `:` and `@`. You can do that in a loop, as long as the error message on failures is clear about what incorrect expression was tried.